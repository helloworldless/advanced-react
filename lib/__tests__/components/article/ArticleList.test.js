import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from '../../../components/article/ArticleList';

describe('ArticleList', () => {

  const mockData = {
    articles: {
      abc: {id: 'abc'},
      def: {id: 'def'}
    },
    articleActions: {
      lookupAuthor: jest.fn(() => ({}))
    }
  };

  const tree = renderer.create(
    <ArticleList {...mockData} />
  ).toJSON();

  it('matches snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  it('has expect number of children', () => {
    expect(tree.children.length).toBe(2);
  });

});
