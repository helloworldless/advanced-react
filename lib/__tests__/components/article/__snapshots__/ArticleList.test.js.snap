// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList matches snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": 600,
          "padding": 10,
        }
      }
    />
    <div
      style={
        Object {
          "color": "gray",
          "padding": 10,
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "padding": 10,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": 600,
          "padding": 10,
        }
      }
    />
    <div
      style={
        Object {
          "color": "gray",
          "padding": 10,
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "padding": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "padding": 10,
        }
      }
    />
  </div>
</div>
`;
